lean-repo-utils.php

```diff
@@ -49,16 +49,349 @@ function pantheon_get_secrets($bindingDir, $requiredKeys, $defaultValues) {
+
+    // The name of the PR branch
+    $branch = $buildMetadata['ref'];
+    // When working from HEAD, use branch master.
+    if ($branch == 'HEAD') {
+        $branch = 'master';
+    }
+
+    // The commit to cherry-pick
+    $commitToSubmit = exec("git -C $fullRepository rev-parse HEAD");
+
+    // Seatbelts: is build metadatafile modified in the HEAD commit?
+    $commitWithBuildMetadataFile = exec("git -C $fullRepository log -n 1 --pretty=format:%H -- $buildMetadataFile");
+    if ($commitWithBuildMetadataFile == $commitToSubmit) {
+        print "Ignoring commit because it contains build assets.\n";
+        return;
+    }
+
+    // A working branch to make changes on
+    $targetBranch = $branch;
+
+    print "::::::::::::::::: Info :::::::::::::::::\n";
+    print "We are going to check out $branch from {$buildMetadata['url']}, branch from $fromSha and cherry-pick $commitToSubmit onto it\n";
+
+    $canonicalRepository = "$workDir/scratchRepository";
+    $workbranch = "recommit-work";
+
+    // Make a working clone of the Git branch. Clone just the branch
+    // and commit we need.
+    passthru("git clone $upstreamRepoWithCredentials --depth=1 --branch $branch --single-branch $canonicalRepository 2>&1");
+
+    // If there have been extra commits, then unshallow the repository so that
+    // we can make a branch off of the commit this multidev was built from.
+    print "git rev-parse HEAD\n";
+    $remoteHead = exec("git -C $canonicalRepository rev-parse HEAD");
+    if ($remoteHead != $fromSha) {
+        // TODO: If we had git 2.11.0, we could use --shallow-since with the date
+        // from $buildMetadata['commit-date'] to get exactly the commits we need.
+        // Until then, though, we will just `unshallow` the whole branch if there
+        // is a conflicting commit.
+        print "git fetch --unshallow\n";
+        passthru("git -C $canonicalRepository fetch --unshallow 2>&1");
+    }
+
+    // Get metadata from the commit at the HEAD of the full repository
+    $comment = escapeshellarg(exec("git -C $fullRepository log -1 --pretty=\"%s\""));
+    $commit_date = escapeshellarg(exec("git -C $fullRepository log -1 --pretty=\"%at\""));
+    $author_name = exec("git -C $fullRepository log -1 --pretty=\"%an\"");
+    $author_email = exec("git -C $fullRepository log -1 --pretty=\"%ae\"");
+    $author = escapeshellarg("$author_name <$author_email>");
+
+    print "Comment is $comment and author is $author and date is $commit_date\n";
+    // Make a safe space to store stuff
+    $safe_space = "$workDir/safe-space";
+    mkdir($safe_space);
+
+    // If there are conflicting commits, or if this new commit is on the master
+    // branch, then we will work from and push to a branch with a different name.
+    // The user should then create a new PR, and use the Git Provider UI to resolve
+    // any conflicts (or clone the branch locally to do the same thing).
+    $createNewBranchReason = '';
+    if ($branch == 'master') {
```

Comment:
`master` is not always the default branch for a repository. Determining the default branch for the repository and protecting it, rather than hard-coding `master`, would be ideal